### ISSUE TITLE
Some Datadog APM services does not list after upgrading ddtrace to 1.0

### ISSUE THREAD

**Issue by kangaechu** [DataDog/dd-trace-rb#2047]
I upgraded ddtrace from 0.47.0 to 1.0.0.
After that, Some services does not list to Datadog APM Service List.
https://app.datadoghq.com/apm/services

Services list in both 0.47.0 and 1.0.0

- mysql2
- shoryuken
- net/http
- elasticsearch
- memcached
- faraday
- active_support-cache
- aws
- rake
- puma
- redis

Services list in 0.47.0, but not list in 1.0.0.

- [SERVICE_NAME]
- delayed_job
- [SERVICE_NAME]-mysql2
- active_record
- [SERVICE_NAME]-cache
- delayed_job-client

Configuration in 0.47.0.

```ruby
Datadog.configure do |c|
  c.analytics_enabled = true

  c.use :rails, analytics_enabled: nil
  c.use :aws, analytics_enabled: nil
  c.use :dalli, analytics_enabled: nil
  c.use :delayed_job, analytics_enabled: nil
  c.use :elasticsearch, analytics_enabled: nil
  c.use :faraday, analytics_enabled: nil
  c.use :http, analytics_enabled: nil
  c.use :mysql2, analytics_enabled: nil
  c.use :redis, analytics_enabled: nil
  c.use :shoryuken, analytics_enabled: nil
end
```

Configuration in 1.0.0.

```ruby
Datadog.configure do |c|
  c.tracing.analytics.enabled = true

  c.tracing.instrument :rails, analytics_enabled: nil
  c.tracing.instrument :aws, analytics_enabled: nil
  c.tracing.instrument :dalli, analytics_enabled: nil
  c.tracing.instrument :delayed_job, analytics_enabled: nil
  c.tracing.instrument :elasticsearch, analytics_enabled: nil
  c.tracing.instrument :faraday, analytics_enabled: nil
  c.tracing.instrument :http, analytics_enabled: nil
  c.tracing.instrument :mysql2, analytics_enabled: nil
  c.tracing.instrument :redis, analytics_enabled: nil
  c.tracing.instrument :shoryuken, analytics_enabled: nil
end
```

Only the following message was output in the Rails log.
(Security-related values are substituted. If you need to these value, I'll send via Datadog support)

```
"I, [2022-05-26T08:44:47.309995 #1]  INFO -- ddtrace: [ddtrace] DATADOG CONFIGURATION - {""date"":""2022-05-26T08:44:47+09:00"",""os_name"":""x86_64-pc-linux-gnu"",""version"":""1.0.0"",""lang"":""ruby"",""lang_version"":""x.y.z"",""enabled"":true,""service"":""puma"",""agent_url"":""http://[IP_ADDRESS]?[PORT]=30"",""debug"":false,""analytics_enabled"":true,""sample_rate"":0.05,""runtime_metrics_enabled"":false,""integrations_loaded"":""rails@x.y.z,aws@x.y.z,dalli@x.y.z,delayed_job@x.y.z,elasticsearch@x.y.z,faraday@x.y.z,http@x.y.z,mysql2@x.y.z,redis@x.y.z,shoryuken@x.y.z,rack@x.y.z,action_cable@x.y.z,action_mailer@x.y.z,active_support@x.y.z,action_pack@x.y.z,action_view@x.y.z,active_job@x.y.z,active_record@x.y.z,lograge@x.y.z"",""vm"":""ruby-x.y.z"",""partial_flushing_enabled"":false,""priority_sampling_enabled"":false,""health_metrics_enabled"":false,""integration_rails_analytics_enabled"":"""",""integration_rails_analytics_sample_rate"":""1.0"",""integration_rails_enabled"":""true"",""integration_rails_service_name"":"""",""integration_rails_distributed_tracing"":""true"",""integration_rails_exception_controller"":"""",""integration_rails_middleware"":""true"",""integration_rails_middleware_names"":""false"",""integration_rails_template_base_path"":""views/"",""integration_aws_analytics_enabled"":"""",""integration_aws_analytics_sample_rate"":""1.0"",""integration_aws_enabled"":""true"",""integration_aws_service_name"":""aws"",""integration_dalli_analytics_enabled"":"""",""integration_dalli_analytics_sample_rate"":""1.0"",""integration_dalli_enabled"":""true"",""integration_dalli_service_name"":""memcached"",""integration_delayed_job_analytics_enabled"":"""",""integration_delayed_job_analytics_sample_rate"":""1.0"",""integration_delayed_job_enabled"":""true"",""integration_delayed_job_service_name"":"""",""integration_delayed_job_client_service_name"":"""",""integration_delayed_job_error_handler"":""#\u003cProc:0x0000556ffeca0e18 /usr/local/bundle/gems/ddtrace-1.0.0/lib/datadog/tracing/span_operation.rb:341\u003e"",""integration_elasticsearch_analytics_enabled"":"""",""integration_elasticsearch_analytics_sample_rate"":""1.0"",""integration_elasticsearch_enabled"":""true"",""integration_elasticsearch_service_name"":""elasticsearch"",""integration_elasticsearch_quantize"":""{}"",""integration_faraday_analytics_enabled"":"""",""integration_faraday_analytics_sample_rate"":""1.0"",""integration_faraday_enabled"":""true"",""integration_faraday_service_name"":""faraday"",""integration_faraday_distributed_tracing"":""true"",""integration_faraday_error_handler"":""#\u003cProc:0x0000556fff7bdaa8 /usr/local/bundle/gems/ddtrace-1.0.0/lib/datadog/tracing/contrib/faraday/configuration/settings.rb:14 (lambda)\u003e"",""integration_faraday_split_by_domain"":""false"",""integration_http_analytics_enabled"":"""",""integration_http_analytics_sample_rate"":""1.0"",""integration_http_enabled"":""true"",""integration_http_service_name"":""net/http"",""integration_http_distributed_tracing"":""true"",""integration_http_split_by_domain"":""false"",""integration_mysql2_analytics_enabled"":"""",""integration_mysql2_analytics_sample_rate"":""1.0"",""integration_mysql2_enabled"":""true"",""integration_mysql2_service_name"":""mysql2"",""integration_redis_analytics_enabled"":"""",""integration_redis_analytics_sample_rate"":""1.0"",""integration_redis_enabled"":""true"",""integration_redis_service_name"":""redis"",""integration_redis_command_args"":""true"",""integration_shoryuken_analytics_enabled"":"""",""integration_shoryuken_analytics_sample_rate"":""1.0"",""integration_shoryuken_enabled"":""true"",""integration_shoryuken_service_name"":"""",""integration_shoryuken_error_handler"":""#\u003cProc:0x0000556ffeca0e18 /usr/local/bundle/gems/ddtrace-1.0.0/lib/datadog/tracing/span_operation.rb:341\u003e"",""integration_shoryuken_tag_body"":""false"",""integration_rack_analytics_enabled"":"""",""integration_rack_analytics_sample_rate"":""1.0"",""integration_rack_enabled"":""true"",""integration_rack_service_name"":"""",""integration_rack_application"":""#\[SERVICE_NAME]::Application:0x0000557001f3b788\u003e"",""integration_rack_distributed_tracing"":""true"",""integration_rack_headers"":""{:response=\u003e[\""Content-Type\"", \""X-Request-ID\""]}"",""integration_rack_middleware_names"":""false"",""integration_rack_quantize"":""{}"",""integration_rack_request_queuing"":""false"",""integration_rack_web_service_name"":""web-server"",""integration_action_cable_analytics_enabled"":""false"",""integration_action_cable_analytics_sample_rate"":""1.0"",""integration_action_cable_enabled"":""true"",""integration_action_cable_service_name"":"""",""integration_action_mailer_analytics_enabled"":""false"",""integration_action_mailer_analytics_sample_rate"":""1.0"",""integration_action_mailer_enabled"":""true"",""integration_action_mailer_service_name"":"""",""integration_action_mailer_email_data"":""false"",""integration_active_support_analytics_enabled"":""false"",""integration_active_support_analytics_sample_rate"":""1.0"",""integration_active_support_enabled"":""true"",""integration_active_support_service_name"":"""",""integration_active_support_cache_service"":""active_support-cache"",""integration_action_pack_analytics_enabled"":"""",""integration_action_pack_analytics_sample_rate"":""1.0"",""integration_action_pack_enabled"":""true"",""integration_action_pack_service_name"":"""",""integration_action_pack_exception_controller"":"""",""integration_action_view_analytics_enabled"":""false"",""integration_action_view_analytics_sample_rate"":""1.0"",""integration_action_view_enabled"":""true"",""integration_action_view_service_name"":"""",""integration_action_view_template_base_path"":""views/"",""integration_active_job_analytics_enabled"":""false"",""integration_active_job_analytics_sample_rate"":""1.0"",""integration_active_job_enabled"":""true"",""integration_active_job_service_name"":"""",""integration_active_job_error_handler"":""#\u003cProc:0x0000556ffeca0e18 /usr/local/bundle/gems/ddtrace-1.0.0/lib/datadog/tracing/span_operation.rb:341\u003e"",""integration_active_record_analytics_enabled"":""false"",""integration_active_record_analytics_sample_rate"":""1.0"",""integration_active_record_enabled"":""true"",""integration_active_record_service_name"":""mysql2"",""integration_lograge_analytics_enabled"":""false"",""integration_lograge_analytics_sample_rate"":""1.0"",""integration_lograge_enabled"":""true"",""integration_lograge_service_name"":""""}"
```

**Comment by delner** [comment]
Hey @kangaechu ! Thanks for the report.

If I'm reading this correctly, this is largely by design, as outlined in our [1.0 upgrade guide](https://github.com/DataDog/dd-trace-rb/blob/master/docs/UpgradeGuide.md#1.0-instrumentation-service-naming).

Reason being is that spans measuring operations done entirely within the Ruby application belong to the same service (whatever `c.service` or `DD_SERVICE` is set to.) Previous to 1.0, we erroneously allowed different service names to be configured for these internal spans: this is corrected in 1.0 and was flagged as a breaking change.

My recommendation is to make sure you set `c.service`/`DD_SERVICE` to a value that's a good name for your application: all internal spans should then automatically reflect this name. That should give you optimal service naming. If service names still don't look right, then we can take another look and figure it out.

Regarding the output message, you can ignore that. It's some startup debug statement that gives some information about the state of the tracer when its started.

**Comment by kangaechu** [comment]
Thank you for replying.
After I set `c.service`. services displayed correctly.
